arrow-core-data/src/main/kotlin/arrow/core/Iterable.kt

```diff
@@ -67,13 +68,34 @@ fun <E, A> Iterable<Validated<E, Iterable<A>>>.flatSequenceValidated(semigroup:
 fun <E> Iterable<Validated<E, *>>.sequenceValidated_(semigroup: Semigroup<E>): Validated<E, Unit> =
   traverseValidated_(semigroup, ::identity)
 
+@Deprecated(
+  "In favor of zip from Kotlin std, please use zip instead of map2",
+  ReplaceWith(
+    "zip(fb, f)"
+  )
+)
 inline fun <A, B, Z> Iterable<A>.map2(fb: Iterable<B>, f: (Tuple2<A, B>) -> Z): List<Z> =
   flatMap { a ->
     fb.map { b ->
       f(Tuple2(a, b))
     }
   }
 
+@Deprecated(
+  "In favor of zip from Kotlin std, please use zip instead of product",
+  ReplaceWith(
+    "zip(fb)"
+  )
+)
+fun <A, B> Iterable<A>.product(fb: Iterable<B>): List<Tuple2<A, B>> =
+  fb.ap(map { a: A -> { b: B -> Tuple2(a, b) } })
+
+inline fun <A, B, Z> Iterable<A>.zip(fb: Iterable<B>, f: (A, B) -> Z): List<Z> =
+  _zip(fb, f)
+
+fun <A, B> Iterable<A>.zip(fb: Iterable<B>): List<Pair<A, B>> =
+  _zip(fb)
```

Comment:
These can be removed since they already exist in `kotlin.collections.zip`. No need to duplicate them under `arrow.core.zip`